NodeJS/src/AgentClient.ts

```diff
@@ -0,0 +1,413 @@
+            subAccountId,
+            agent_username,
+            agent_password,
+            agentStationGroupId,
+            agentLoginId,
+            skillIds, skillsWithPriority
+        );
+
+        // wait until agent is created
+        await this.waitForAgentCreation(agentLoginId);
+
+        let stationExtension = await this.generateExtension(subAccountId, 'STATION');
+        await this.sendCreateStationRequest(
+            agentStationGroupId,
+            subAccountId,
+            stationExtension,
+            agent_username
+        );
+
+        // wait until station is created
+        await this.waitForStationCreation(subAccountId, agent_username);
+        return this.getAgent(agent_username);
+    }
+
+    async getAgentStationGroupId(subAccountId: string) {
+        return this.session.get('/spokenAbc/agentStationGroups/client/' + subAccountId)
+            .then((response: { data: any }) => {
+                let agentStationGroups = response.data;
+                // ensure we always get the same subAccount ordering
+                agentStationGroups = lodash.sortBy(agentStationGroups, ['id']);
+                return agentStationGroups[0].id;
+            })
+    }
+
+    async sendCreateAgentRequest(
+        subAccountId: any,
+        agent_username: string,
+        agent_password: string,
+        agentStationGroupId: any,
+        agentLoginId: any,
+        skillIds: any, skillsWithPriority:[SkillPriority]) {
+
+        let securityCode = this.generateSecurityCode(agentLoginId);
+        let avayaPassword = this.generateAvayaPassword(agentLoginId);
+        let agent = {
+            "username": agent_username,
+            "firstName": AGENT_FIRST_NAME,
+            "lastName": AGENT_LAST_NAME,
+            "password": agent_password,
+            "loginId": agentLoginId,
+            "agentStationGroupId": agentStationGroupId,
+            "securityCode": securityCode,
+            "startDate": "2019-03-21",
+            "endDate": "2038-01-01",
+            "avayaPassword": avayaPassword,
+            "clientId": subAccountId,
+            "skillIds": skillIds,
+            "agentSkills": skillsWithPriority,
+            // no supervisors
+            "supervisorId": 0,
+            // channel 1 is voice
+            "channelIds": [1]
+        };
+
+        return this.session.post('/spokenAbc/jobs/agents', agent)
+            .then((result: any) => {
+                return result
+            })
+    }
+
+    generateAvayaPassword(agentLoginId: { toString: () => any }) {
+        let agentLoginIdString = agentLoginId.toString();
+        let length = agentLoginIdString.length;
+        return agentLoginIdString.substring(length - 6, length);
+    }
+
+    generateSecurityCode(agentLoginId: { toString: () => any }) {
+        let agentLoginIdString = agentLoginId.toString();
+        let length = agentLoginIdString.length;
+        return agentLoginIdString.substring(length - 4, length);
```

Comment:
same like above.   generateSecurityCode should be a method of agentLoginId.  Same for the above method.